src/kbmod/filters/sns_filters.py

```diff
@@ -36,3 +37,98 @@ def peak_offset_filter(res, peak_offset_max=6):
     radial_d = ((X - b / 2) ** 2 + (Y - a / 2) ** 2) ** 0.5
     w = np.where(radial_d < peak_offset_max)
     res.table = res.table[w]
+
+
+def predictive_line_cluster(res, dmjds, dist_lim=4.0, min_samp=2, init_select_proc_distance=60):
+    """???
+
+    Parameters
+    ----------
+    res : `Results`
+        The search results containing trajectories.
+        This object is modified by filtering out rows.
+    dmjds : `list`
+        List of dates in dmjd format.
+    dist_lim : `double`
+    min_samp : `int`
+    init_select_proc_distance : `int`
+    """
+
+    snr = res["psi_curve"] / np.sqrt(res["phi_curve"])
+    max_snr = np.max(snr, axis=1)
+
+    x_col = res["x"].data
+    y_col = res["y"].data
+    vx_col = res["vx"].data
+    vy_col = res["vy"].data
+    res_copy = res
+
+    proc_inds = np.arange(len(x_col))
+    clust_inds = []
+
+    while len(max_snr) > 0:
+        arg_max = np.argmax(max_snr)
+        x_o = x_col[arg_max]
+        y_o = y_col[arg_max]
+        rx_o = vx_col[arg_max]
+        ry_o = vy_col[arg_max]
+
+        # this secondary where command is necessary because of memory overflows in large detection lists
+        w = np.where(
+            (x_col > x_o - 55)
+            & (x_col < x_o + init_select_proc_distance)
+            & (y_col > y_o - 55)
+            & (y_col < y_o + init_select_proc_distance)
```

Comment:
heard back from wes! the 55 should actually be the `init_select_proc_distance` value